src/experimental/Toast/ToastUI.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import {CheckIcon, AlertIcon, StopIcon, InfoIcon, XIcon} from '@primer/octicons-react'
+
+import {Spinner, Box, IconButton} from '../..'
+
+const variants = ['default', 'success', 'attention', 'danger', 'loading'] as const
+
+const icon: Record<ToastVariants, React.ReactNode> = {
+  default: <InfoIcon size="small" />,
+  success: <CheckIcon size="small" />,
+  attention: <AlertIcon size="small" />,
+  danger: <StopIcon size="small" />,
+  loading: <Spinner size="small" />,
+}
+
+export const ToastUI = ({
+  dismissable = false,
+  variant = 'default',
+  role = 'status',
+  dismissLabel = 'Dismiss this message',
+  children,
+}: React.PropsWithChildren<ToastUIProps>) => {
+  if (__DEV__) {
+    // eslint-disable-next-line no-console
+    console.assert(
+      variants.includes(variant),
+      `The variant ${variant} is not supported. Use one of ${variants.join(', ')}`,
+    )
+  }
+
+  return (
+    <Box role={role} sx={{display: 'flex', justifyContent: 'flex-start'}}>
+      <Box
+        sx={{display: 'inline-flex', maxWidth: '450px', borderRadius: 2, boxShadow: 'shadow.large', overflow: 'hidden'}}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            justifyItems: 'center',
+            alignItems: 'center',
+            borderTopLeftRadius: 2,
+            borderBottomLeftRadius: 2,
+            border: '1px solid',
+            borderColor: 'border.default',
+            borderInlineEnd: 'none',
+            p: 3,
+            color: 'white',
+            '&[data-variant="default"]': {
+              backgroundColor: 'accent.fg',
+              borderColor: 'accent.muted',
+            },
+            '&[data-variant="success"]': {
+              backgroundColor: 'success.fg',
+              borderColor: 'success.muted',
+            },
+            '&[data-variant="attention"]': {
+              backgroundColor: 'attention.fg',
+              borderColor: 'attention.muted',
+            },
+            '&[data-variant="danger"]': {
+              backgroundColor: 'danger.fg',
+              borderColor: 'danger.muted',
+            },
+            '&[data-variant="loading"]': {
+              backgroundColor: 'neutral.emphasis',
+              borderColor: 'neutral.muted',
+            },
+          }}
+          role="presentation"
+          data-variant={variant}
+        >
+          {icon[variant]}
+        </Box>
+        <Box
+          sx={{
+            display: 'flex',
+            alignItems: 'center',
+            gap: 3,
+            justifyContent: 'space-between',
+            borderBottomRightRadius: 2,
+            borderTopRightRadius: 2,
+            border: '1px solid',
+            borderColor: 'border.default',
+            borderInlineStart: 'none',
+            px: 3,
+          }}
+        >
```

Comment:
```suggestion
        <Box
          sx={{
            display: 'flex',
            alignItems: 'center',
            gap: 3,
            justifyContent: 'space-between',
            borderBottomRightRadius: 2,
            borderTopRightRadius: 2,
            border: '1px solid',
            borderColor: 'border.default',
            borderInlineStart: 'none',
            px: 3,
            paddingRight: dismissible ? 2 : 3,
          }}
        >
```

If `dismissible` is true with the icon button, the padding right should be 8px for visual alignment.